configuration/src/test/java/io/camunda/configuration/SecondaryStorageTest.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.configuration;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.camunda.configuration.beanoverrides.OperatePropertiesOverride;
+import io.camunda.configuration.beanoverrides.TasklistPropertiesOverride;
+import io.camunda.operate.property.OperateProperties;
+import io.camunda.search.connect.configuration.DatabaseType;
+import io.camunda.tasklist.property.TasklistProperties;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit.jupiter.SpringJUnitConfig;
+
+@SpringJUnitConfig({
+  UnifiedConfiguration.class,
+  UnifiedConfigurationHelper.class,
+  TasklistPropertiesOverride.class,
+  OperatePropertiesOverride.class,
+})
+public class SecondaryStorageTest {
+
+  @Nested
+  @TestPropertySource(
+      properties = {
+        "camunda.data.secondary-storage.type=elasticsearch",
+        "camunda.data.secondary-storage.elasticsearch.url=http://expected-url:4321"
+      })
+  class WithOnlyUnifiedConfigSet {
+    final OperateProperties operateProperties;
+    final TasklistProperties tasklistProperties;
+
+    WithOnlyUnifiedConfigSet(
+        @Autowired final OperateProperties operateProperties,
+        @Autowired final TasklistProperties tasklistProperties) {
+      this.operateProperties = operateProperties;
+      this.tasklistProperties = tasklistProperties;
+    }
+
+    // @Test
+    void testCamundaDataSecondaryStorageType() {
+      final DatabaseType expectedOperateDatabaseType = DatabaseType.ELASTICSEARCH;
+      assertThat(operateProperties.getDatabase()).isEqualTo(expectedOperateDatabaseType);
```

Comment:
:x: `DatabaseType` is from the operate module. Here you have imported the one from search-client.